### ISSUE TITLE
[FEATURE] Add `no-aria-hidden-on-focusable` rule

### ISSUE THREAD

**Issue by yeonjuan** [yeonjuan/html-eslint#353]
**Is your feature request related to a problem? Please describe.**
 Add `no-aria-hidden-on-focusable` rule to disallow `aria-hidden="true"` on focusable elements

**correct**
```html
  <div aria-hidden="true" />
  <img aria-hidden="true" />
  <a aria-hidden="false" href="#" />
  <button aria-hidden="true" tabIndex="-1" /> // `tabIndex=-1` removes the element from sequential focus navigation so we don't flag it.
  <a href="/" />
  <div aria-hidden="true"><a href="#"></a></div> // This is also bad but will not be handled by this rule.
```

**incorrect**

```html
  <div aria-hidden="true" tabIndex="0" />
  <input aria-hidden="true" />
  <a href="/" aria-hidden="true" />
  <button aria-hidden="true" />
  <textarea aria-hidden="true" />
```

**Additional context**
- https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-aria-hidden-on-focusable.md

**Comment by System625** [comment]
Hi there! I'm Olamiposi, a Frontend developer with 3+ years of experience building scalable and beautiful web applications.
I will add the rule as instructed following the context in the link that was attached in the description
Would love to contribute to this amazing project

**Comment by yeonjuan** [comment]
Hi @System625 Thanks! If you have any questions along the way, feel free to reach out anytime.
